src/lib/components/Instances/GenerateInstances.svelte

```diff
@@ -0,0 +1,68 @@
+<script lang="ts">
+	import { clickOutside } from '$lib/clickOutside';
+	import Button from '$lib/components/ui/Button.svelte';
+	import { Sparkle, Sparkles } from 'lucide-svelte';
+	import { fade } from 'svelte/transition';
+
+	let { similar, createInstances } = $props<{
+		similar: boolean;
+		createInstances: (instruction: string, count: number, similar: boolean) => void;
+	}>();
+
+	let showGenerateOptions = $state(false);
+	let generationInstruction = $state('');
+	let generationCount = $state(5);
+</script>
+
+<div class="relative" use:clickOutside={() => (showGenerateOptions = false)}>
```

Comment:
nit: if I drag the slider and lift the mouse outside it closes. Maybe we should check for mouse down in this case?

Reply:
This doesn't happen for me I don't think? Like you keep it pressed, move outside of the popup, and lift?

Reply:
Yep, but in that case maybe we fix it later.